### DISCUSSION TITLE
Linking with 3rd party libraries

### DISCUSSION THREAD

**Discussion by adperezm** [ExtremeFLOW/neko#566] [Q&A] [answered]
Is there a standard procedure to link Neko to an external library?

In my specific case, I need to link Neko with ADIOS2 to integrate in-situ compression. If there are no instructions on how to do this, an example with one of the libraries currently used would be of help.

**Comment by njansson** [comment, marked as answer]
The procedure would be to find the library during configure by writing a `.m4` file (e.g. https://github.com/ExtremeFLOW/neko/blob/develop/m4/ax_metis.m4), and add the libraries to the link flags.

**Reply by njansson** [reply]
A simpler example can be found in https://github.com/ExtremeFLOW/neko/blob/develop/m4/ax_darshan.m4

**Reply by njansson** [reply]
@adperezm, Link flags for ADIOS2 has been added in #567, just configure with,
 `--with-adios2=/path/to/adios2` and it should cxx and link flags from `adios2-config` 🍻, if you are interested in how Neko finds adios, have a look at `m4/ax_adios.m4`

To add ADIOS2 related C++ code, please have a look at the comment in `src/Makefile.am` close to the `if ENABLE ADIOS2` line.

**Reply by adperezm** [reply]
Great, thank you!
